### PR TITLE
Add UnitStep ....

### PR DESCRIPTION
**Pull request by rocky**
Also, refactor mathics.eval.arithmetic to remove eval functions belonging to mathics.builtin.arithfns, and mathics.builtin.numeric

### CODE REVIEW

File: mathics/builtin/numeric.py [hunk 1]
```diff
@@ -787,3 +790,48 @@ def eval(self, x, evaluation: Evaluation):
...
+    :Heaviside step function:
+    https://en.wikipedia.org/wiki/Heaviside_step_function</url> (<url>
+    :WMA link:
+    https://reference.wolfram.com/language/ref/UnitStep.html</url>)
+
+    <dl>
+      <dt>'UnitStep[$x$]'
+      <dd>return 0 if $x$ < 0, and 1 if $x$ >= 0.
+      <dt>'UnitStep[$x1$, $x2$, ...]'
+      <dd>return the multidimenional unit step function which is 1 only if none of the $xi$ are negative.
```
Comment (axkr) [inline mathics/builtin/numeric.py:807]: typo `multidimensional`

File: mathics/builtin/numeric.py [hunk 2]
```diff
@@ -787,3 +790,48 @@ def eval(self, x, evaluation: Evaluation):
...
+    </dl>
+
+    Evaluation numerically:
+    >> UnitStep[0.7]
+     = 1
+
+    We can use 'UnitStep' on irrational numbers:
+    >> Table[UnitStep[x], {x, -3, 3}]
+     = {1, 1, 0}
```
Comment (axkr) [inline mathics/builtin/numeric.py:816]: should return `{0,0,0,1,1,1,1}`

File: mathics/builtin/numeric.py (hunk 1 above)
Comment (rocky) [reply mathics/builtin/numeric.py]: Thanks! Fixed now. (I should try to make codespell part of my workflow better...)

File: mathics/builtin/numeric.py (hunk 2 above)
Comment (rocky) [reply mathics/builtin/numeric.py]: Ooops. Was in a rush to commit. Have gone over things better now. Thanks for catching.

Comment (rocky) [reply mathics/builtin/numeric.py:816]: I see what happened that output was intended for the missing example: `Map[UnitStep, {Pi, Infinity, -Infinity}]`

**Comment by rocky** [comment]
With this change "BoolEval.m" starts to work.

**Review by mmatera — APPROVED** [review]
LGTM

**Pull request merged**